cypress-io/cypress
# feat: run failures enhancements

- Closes #24854
### User facing changelog
na

### Additional details
Working with Ankit revealed we needed to change up `DebugMappings.ts`. We paired on this and the results of the refactor are relevant for his upcoming work (groups).

### Steps to test
Driven mostly from CT tests, if you want to run it e2e:

1. Check out the `feature/CYCLOUD-665-IATR` and run `yarn`
2. Run `yarn docker-compose && yarn dev`
3. Visit `localhost:3000` and create a project
4. In the Cypress monorepo, checkout this branch and run `CYPRESS_INTERNAL_ENV=development yarn cypress:open`
5. Select a project and add your newly created `projectId`
6. Tweak tests to cause a failure and run `npx cypress run --record --key <RECORD_KEY>`
7. If you've recorded different runs, you can change the run by changing `runByNumber(runNumber: 6) {` to whatever run you want to target (not dynamic yet)

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/208729758-53a8565f-cdbd-40ea-af6f-dba66ce50256.mov

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
